src/librustdoc/test.rs

```diff
@@ -40,12 +40,26 @@ pub struct TestOptions {
     pub attrs: Vec<String>,
 }
 
+macro_rules! try_none {
+    ($x:expr) => {
+        match $x {
+            Some(y) => y,
+            None => return None,
+        }
+    }
+}
+
 pub fn run(options: Options) -> i32 {
     let input = config::Input::File(options.input.clone());
 
     let sessopts = config::Options {
-        maybe_sysroot: options.maybe_sysroot.clone().or_else(
-            || Some(env::current_exe().unwrap().parent().unwrap().parent().unwrap().to_path_buf())),
+        maybe_sysroot: options.maybe_sysroot.clone().or_else(|| {
+            Some(try_none!(
+                try_none!(
+                    try_none!(env::current_exe().ok()).parent()
+                ).parent()
+            ).to_path_buf())
+        }),
```

Comment:
This should just be `maybe_sysroot: options.maybe_sysroot.clone(),` which will let rustc find the sysroot if it wasn't passed as a command line attribute. Unless there's some reason rustdoc needs to call `current_exe` itself but I can't see any.